src/App.tsx

```diff
@@ -1,223 +1,167 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext, useEffect, useState,
+} from 'react';
+import {
+  deleteTodo, getCompletedTodos, getTodos, postTodo,
+} from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import { ErrorMessage } from './components/ErrorMessage';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
+import { TodosList } from './components/TodosList';
+import { ErrorType } from './types/Error';
+import { FilterStatus } from './types/FilterStatus';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
-  const newTodoField = useRef<HTMLInputElement>(null);
+  const [visibleToDos, setVisibleToDos] = useState<Todo[]>([]);
+  const [hasCompleted, setHasCompleted] = useState(false);
+  const [activeCount, setActiveCount] = useState(0);
+  const [filterStatus, setFilterStatus] = useState(FilterStatus.All);
+  const [errorType, setErrorType] = useState<ErrorType | null>(null);
+  const [requestCount, setRequestCount] = useState(0);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [isAdding, setIsAdding] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [deletingToDoId, setDeletingToDoId] = useState<number[]>([]);
+
+  const handleErrorClose = () => setErrorType(null);
+
+  const getSelectedTodos = (todos: Todo[]) => {
+    const filteredTodos = todos.filter(({ completed }) => {
+      switch (filterStatus) {
+        case FilterStatus.Active:
+          return !completed;
+
+        case FilterStatus.Completed:
+          return completed;
+
+        default:
+          return true;
+      }
+    });
+
+    setVisibleToDos(filteredTodos);
```

Comment:
do not store filtered todos in state - calculate them and store in value instead (use useMemo to improve performance)